include/ignition/rendering/BoundingBoxCamera.hh

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (C) 2021 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+#ifndef IGNITION_RENDERING_BoundingBoxCAMERA_HH_
+#define IGNITION_RENDERING_BoundingBoxCAMERA_HH_
+
+#include <string>
+#include <vector>
+#include <cstdint>
+
+#include <ignition/common/Event.hh>
+#include "ignition/rendering/Camera.hh"
+
+
+namespace ignition
+{
+  namespace rendering
+  {
+    /// \brief BoundingBox types for Visible / Full BoundingBox
+    /// FullBox: Shows the full box of occluded objects
+    /// VisibleBox: Shows the visible part of the occluded object
+    enum BoundingBoxType {
+      FullBox,
+      VisibleBox
+    };
+
+    /// \brief Bounding box structure to store its boundaries
+    /// min & max of x & y, and the label
+    struct BoundingBox
```

Comment:
You could use `ignition::math::AxisAlignedBox` instead of creating a new type. That's meant for 3D boxes, but I think it should be ok to leave the Z dimension as zero when doing 2D :thinking:

Reply:
@chapulina 
It only provides the bounding boxes, that RGB image because in the example I have 2 cameras, and I draw the bboxes on the image of the rgb camera (just for seek of visualization and trying the example)

mmmm, yeah I have a plan to see how to provide the 3d boxes with the lidar data, but that after I finish all thing in the project :D .. do you think we should support 3d boxes with images instead of lidar ?

about the `math::AxisAlignedBox ` .. I created a new struct because I had to store the label of the bounding box with the dimensions

Reply:
> It only provides the bounding boxes, that RGB image because in the example I have 2 cameras,

Ah perfect!

> do you think we should support 3d boxes with images instead of lidar ?

We have access to the ground truth from simulation, so I think we could just get the 3D bounding box of the visuals that are within a given volume (like a camera frustum?). If we want to take occlusion into account, I think we need to define the observation point, like a camera's image plane.

> I created a new struct because I had to store the label of the bounding box with the dimensions

Yeah I noticed that. You could use the `AxisAlignedBox` inside your struct. The advantage is that the class offers various helper functions that could be useful for users.

Reply:
That is because this PR is done before doing the msgs PR
But it is much simpler to use the struct instead of the msg, as i pass this class everywhere in the code
can we just keep it ?